samples/simulation/qaoa/QAOA.qs

```diff
@@ -0,0 +1,240 @@
+namespace Microsoft.Quantum.Samples.QAOA {
+    open Microsoft.Quantum.Intrinsic;
+    open Microsoft.Quantum.Canon;
+    open Microsoft.Quantum.Measurement;
+    open Microsoft.Quantum.Convert;
+    open Microsoft.Quantum.Arrays;
+    open Microsoft.Quantum.Math;
+    open Microsoft.Quantum.Diagnostics;
+
+    /// # Summary
+    /// This operation applies the X-rotation to each qubit. We can think of it as time 
+    /// evolution induced by applying H = - \sum_i X_i for time t.
+    ///
+    /// # Input
+    /// ## target
+    /// Target qubit register
+    /// ## time
+    /// Time passed in evolution of X rotation
+    operation DriverHamiltonian(target: Qubit[], time: Double) : Unit {
+        for(qubit in target)
+        {
+            Rx(-2.0 * time, qubit);
+        }
+    }
+
+    /// # Summary
+    /// This applies the Z-rotation according to the instance Hamiltonian. 
+    /// We can think of it as Hamiltonian time evolution for time t induced
+    /// by the Ising Hamiltonian $\sum_ij J_ij Z_i Z_j + \sum_i h_i Z_i$.
+    ///
+    /// # Input
+    /// ## target
+    /// Qubit register that encodes the Spin values in the Ising Hamiltonian
+    /// ## time
+    /// Time point in evolution
+    /// ## weights
+    /// Ising magnetic field or "weights" encoding the constraints of our
+    /// traveling Santa problem.
+    /// ## coupling
+    /// Ising coupling term or "penalty" encoding the constraints of our
+    /// traveling Santa problem.
+    operation InstanceHamiltonian(
+        target: Qubit[], 
+        time: Double, 
+        weights: Double[], 
+        coupling: Double[]
+    ) : Unit {
+        using (auxiliary = Qubit())
+        {
```

Comment:
[formatting nit] Generally, Q# code uses braces at the end of each line, by contrast with C#, which uses braces on the next line.
```suggestion
        using (auxiliary = Qubit()) {
```